### PR TITLE
[CH-383] Fix 'Unknown identifier' exception

### PR DESCRIPTION
**Pull request by exmy**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

 #383

### CODE REVIEW

**Comment by kyligence-git** [comment]
Can one of the admins verify this patch?

**Comment by liuneng1994** [comment]
test this please

File: utils/local-engine/Parser/SerializedPlanParser.cpp [hunk 1]
```diff
@@ -2553,15 +2553,19 @@ void SerializedPlanParser::reorderJoinOutput(QueryPlan & plan, DB::Names cols)
...
-        const auto & node = actionsDag->addFunction(function_builder, args, item);
-        actionsDag->addOrReplaceInOutputs(node);
+        const auto * require_node = actionsDag->tryFindInOutputs(item);
+        if (require_node)
+        {
+            auto function_builder = FunctionFactory::instance().get("assumeNotNull", context);
+            ActionsDAG::NodeRawConstPtrs args = {require_node};
+            const auto & node = actionsDag->addFunction(function_builder, args, item);
+            actionsDag->addOrReplaceInOutputs(node);
+        }
```
Comment (liuneng1994) [inline utils/local-engine/Parser/SerializedPlanParser.cpp:2568]: 为什么会出现require的列不存在于actionDag的output中呢

Comment (exmy) [reply utils/local-engine/Parser/SerializedPlanParser.cpp:2568]: 这种 case：`where event.log_extra['abflags_v2'] is not null and hdid is not null and hdid != ''`,
 output 是 `and(and(and(isNotNull(event),isNotNull(arrayElement(tupleElement(event,1_0),abflags_v2_1))),isNotNull(hdid)),not(equals(hdid,_2)))` ,
`arrayElement(tupleElement(event,1_0),abflags_v2_1))` 是中间结果，不在 output 中

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Pull request merged**